### DISCUSSION TITLE
Workflow checkpointing support

### DISCUSSION THREAD

**Discussion by bpatzak** [mupif/mupif#62] [Ideas]
Consider adding checkpointing support for (long-running) workflows.
Allow to store workflow state at specific (user-defined) points, saving the workflow state and restart later. Would require the state save/restore support from individual models.

**Comment by eudoxos** [comment]
This looks like quite a task. What I did once (with WooDEM.org) was to compute md5 hash from input data and periodically save the state to file containing the md5 in its name. The simulation would check at start whether some state already exists and would load it. I don't think this approach would be directly usable, since jobmanager will delete temporary (non-output) files after the model quits; plus saving state will be necessarily very much model-specific.

A middle ground could be to save state in-between models within one workflow, so that restarting the workflow could re-use outputs of the models which already completed. Is this what you had in mind?
